plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Vesting.hs

```diff
@@ -126,7 +133,8 @@ retrieveFunds vs vd r vnow = do
         vd' = vd {vestingDataPaidOut = vnow `Value.plus` vestingDataPaidOut vd }
         inp = scriptTxIn r val redeemerScript
         range = W.intervalFrom currentSlot
-    _ <- createTxAndSubmit range (Set.singleton inp) [oo, o]
+    (fee, change) <- createPaymentWithChange transactionFee
+    _ <- createTxAndSubmit range (Set.insert inp fee) ([oo, o] ++ maybeToList change)
```

Comment:
I moderately prefer `Foldable.toList` over `maybeToList`.

Reply:
I can change it, but I think `maybeToList` is easier to parse for someone just starting with Haskell.

Reply:
I don't think we need to optimize the use cases for *total* beginners. Besides, I think `toList` is obvious enough.